### PR TITLE
rpc: concurrent map issue with log subscription fix

### PR DESCRIPTION
**Pull request by hexoscott**
This time the issue was replicated in the test first to confirm what the problem was.

Opted to clone the maps and return them whilst the mutex is held rather than use the mutex from within the call to `SubscribeLogs` and `UnsubscribeLogs` directly as it seemed to be a little hacky, although likely more performant.  Happy to change the code to suit if that's a problem.

Created to solve #4893

### CODE REVIEW

File: turbo/rpchelper/logsfilter.go [hunk 1]
```diff
@@ -94,6 +94,23 @@ func (a *LogsFilterAggregator) addLogsFilters(f *LogsFilter) {
 	}
 }
 
+func (a *LogsFilterAggregator) getAggMaps() (map[common.Address]int, map[common.Hash]int) {
+	a.logsFilterLock.Lock()
```
Comment (AskAlexSharov) [inline turbo/rpchelper/logsfilter.go:98]: RLock?

File: turbo/rpchelper/filters.go [hunk 2]
```diff
@@ -425,14 +428,17 @@ func (ff *Filters) UnsubscribeLogs(id LogsSubID) bool {
...
+	addresses, topics := ff.logsSubs.getAggMaps()
+
+	for addr := range addresses {
 		lfr.Addresses = append(lfr.Addresses, gointerfaces.ConvertAddressToH160(addr))
 	}
-	for topic := range ff.logsSubs.aggLogsFilter.topics {
+	for topic := range topics {
 		lfr.Topics = append(lfr.Topics, gointerfaces.ConvertHashToH256(topic))
 	}
+	ff.mu.Lock()
```
Comment (AskAlexSharov) [inline turbo/rpchelper/filters.go:440]: Maybe let’s create some method which will do load and delete (or only delete) and move lock inside that method?

**Review by AskAlexSharov — CHANGES_REQUESTED** [review]

**Comment by hexoscott** [comment]
Hi @AskAlexSharov, I've pushed the changes requested there.  Out of curiosity, why the preference for a method to hold the lock?  Is that just to keep it locked for as little time as possible?

**Comment by AskAlexSharov** [comment]
@hexoscott as little, and also because I see there is 2 locks are overlaping - some part of code is protected by 2 locks - it smells.

**Review by AskAlexSharov — APPROVED** [review]

**Pull request merged**